tests/test_deprecations.py

```diff
@@ -98,3 +98,9 @@ def test_deprecated_campaign_tolerance_flag(flag):
     """Constructing a Campaign with the deprecated tolerance flag raises an error."""
     with pytest.raises(DeprecationError):
         Campaign(None, None, None, numerical_measurements_must_be_within_tolerance=flag)
+
+
+def test_deprecated_batch_quantity_keyword(campaign):
+    """Using the deprecated batch_quantity keyword raises an error."""
+    with pytest.raises(DeprecationError):
```

Comment:
in this situation I woudl actually expect the warning, not the error? its just a renaming and not moving to to some other place like with the flag

seems inconsistent with other renamings

Reply:
You see, in this particular case the problem is that there is no foolproof "autofix" that the deprecation mechanism could provide, which would be required if you just raise a warning (in which the code execution would go on and the deprecation mechanism would need to make the necessary modifications). If someone passed for example `batch_quantity=6` to the method, inside the function you'd see the two local variables `batch_quantity=6` AND `batch_size=5`. From that you could conclude that someone used the old interface. However, if for some weird reason (e.g. if the call happens programmatically of something) both arguments would be passed, the situation would be unclear.

I know, rather unlikely, but I think it's not too bad to just stick with the mantra "if autofixable --> raise warning; if something's ambiguous --> raise error". Anyway, it's juste temporary code ... 🙃